portfolio/src/main/webapp/script.js

```diff
@@ -28,3 +28,40 @@ const randomFactGenerator = () => {
     const factContainer = document.getElementById('fact-container');
     factContainer.innerText = randomFact;
 };
+
+/**
+ * Fetches a the content from the server and adds it to the DOM.
+ */
+const getServletContent = () => {
+    console.log('Fetching the content from the server.');
+
+    // The fetch() function returns a Promise because the request is asynchronous.
+    const responsePromise = fetch('/data');
+
+    // When the request is complete, pass the response into handleResponse().
+    responsePromise.then(handleResponse);
+};
+
+/**
+ * Handles response by converting it to text and passing the result to
+ * addQuoteToDom().
+ */
+const handleResponse = response => {
+  console.log('Handling the response.');
+
+  // response.text() returns a Promise, because the response is a stream of
+  // content and not a simple variable.
+  const textPromise = response.text();
+
+  // When the response is converted to text, pass the result into the
+  // addQuoteToDom() function.
+  textPromise.then(addContentToDOM);
+};
+
+/** Adds a random quote to the DOM. */
+const addContentToDOM = content => {
+  console.log('Adding content to dom: ' + content);
+
+  const quoteContainer = document.getElementById('servlet-content');
```

Comment:
(opt) Maybe set this const in `document.onload()`, so we don't have to traverse the DOM on each click?

Reply:
I used a button, so that's why I chose to the onclick

Reply:
Again, similar to the example they gave us. For Step four, I made it so that the comments I fetch from the DataServlet.java is presented in the body on every load of the page.

Reply:
Sorry, I miscommunicated:

I mean that you're saying `const quoteContainer = document.getElementById('servlet-content');` but that element is available on load, and never changes. Traversing the DOM tree of such a small page is cheap, but not _free,_ per se. Why not just define `const quoteContainer` once?